src/hooks/use-search-result-filter.ts

```diff
@@ -0,0 +1,22 @@
+import { DropDownOption } from '@/components/drop-down-options/drop-down-options';
+
+export const useSearchResultFilter = (isCaseSensitive: boolean) => {
+  return { filterSearchResults };
+
+  function filterSearchResults<T>(searchOptions: DropDownOption<T, string>[], searchValue: string) {
+    return searchOptions.filter(({ value }) => resultContainsValue(value, searchValue));
+  }
+
+  function resultContainsValue(result: string, searchValue: string) {
+    result = normalizeString(result);
+    searchValue = normalizeString(searchValue);
+    return result.includes(searchValue);
+  }
+
+  function normalizeString(str: string) {
```

Comment:
is this different from the `normalize` method you added in `string.ts`?

Reply:
no, it's the same. I deleted it here, but forgore to push:sob: